test/parallel_exec.jl

```diff
@@ -11,6 +11,78 @@ elseif Base.JLOptions().code_coverage == 2
 end
 addprocs(3; exeflags=`$cov_flag $inline_flag --check-bounds=yes --depwarn=error`)
 
+# Test remote()
+
+append!(Base.default_worker_pool(), workers())
+
+let
+    pool = Base.default_worker_pool()
+
+    r = nothing
+    count = 0
+    function remote_sleep(t)
+        @async begin
+            count += 1
+            r = remote(sleep)(t)
+            count -= 1
+        end
+        yield()
+    end
+
+    w = workers()
+    @test length(w) == 3
+
+    t1 = now()
+
+    @test isready(pool) == true
+    @test count == 0
+    remote_sleep(10)
+    @test count == 1
+                                        # w1: 10s
+
+    @test isready(pool) == true
+    remote_sleep(2)
+    @test count == 2
+                                        # w1: 10, w2: 2s
+
+    @test isready(pool) == true
+    remote_sleep(4)
+    @test count == 3
+    @test isready(pool) == false
+    remote_sleep(4)
+    @test isready(pool) == false
+                                        # w1: 10s, w2: 2s, w3: 4s
+
+    t2 = now()
+    @test (t2-t1).value < 500
+
+    sleep(3)
+                                        # w1: 10s, w2: 2s, w3: 4s ...
+                                        # w1:  9s, w2: 1s, w3: 3s ...
+                                        # w1:  8s, w2: !!, w3: 2s
+    @test count == 3
+    @test isready(pool) == false
+                                        # w1:  8s, w2: 4s, w3: 2s ...
+                                        # w1:  7s, w2: 3s, w3: 1s ...
+                                        # w1:  6s, w2: 2s, w3: !! ...
+    sleep(2)
+    @test count == 2
+    @test isready(pool) == true
+    sleep(3)                            # w1:  6s, w2: 2s ...
+    @test count == 1
+    @test isready(pool) == true
+                                        # w1:  5s, w2: 1s ...
+                                        # w1:  4s, w2: !!
+                                        # w1:  3s ...
+    sleep(4)                            # w1:  2s ...
+    @test count == 0
+    @test isready(pool) == true
+                                        # w1:  1s ...
+                                        # w1:  !!
+                                        # ...
+end
+
+
```

Comment:
We have had tests hitting upper limits and timing out on CI. Would be better to rewrite these tests with a different strategy so that they do not add 10 seconds to the parallel test run.